packages/mui-material/src/Divider/Divider.js

```diff
@@ -54,100 +55,140 @@ const DividerRoot = styled('div', {
+    {
+      props: {
+        absolute: true,
+      },
+      style: {
+        position: 'absolute',
+        bottom: 0,
+        left: 0,
+        width: '100%',
+      },
+    },
+    {
+      props: {
+        light: true,
+      },
+      style: {
+        borderColor: theme.vars
+          ? `rgba(${theme.vars.palette.dividerChannel} / 0.08)`
+          : alpha(theme.palette.divider, 0.08),
+      },
+    },
+    {
+      props: {
+        variant: 'inset',
+      },
+      style: {
+        marginLeft: 72,
+      },
+    },
+    {
+      props: {
+        variant: 'middle',
+        orientation: 'horizontal',
+      },
+      style: {
         marginLeft: theme.spacing(2),
         marginRight: theme.spacing(2),
-      }),
-    ...(ownerState.variant === 'middle' &&
-      ownerState.orientation === 'vertical' && {
+      },
+    },
+    {
+      props: {
+        variant: 'middle',
+        orientation: 'vertical',
+      },
+      style: {
         marginTop: theme.spacing(1),
         marginBottom: theme.spacing(1),
-      }),
-    ...(ownerState.orientation === 'vertical' && {
-      height: '100%',
-      borderBottomWidth: 0,
-      borderRightWidth: 'thin',
-    }),
-    ...(ownerState.flexItem && {
-      alignSelf: 'stretch',
-      height: 'auto',
-    }),
-  }),
-  ({ ownerState }) => ({
-    ...(ownerState.children && {
-      display: 'flex',
-      whiteSpace: 'nowrap',
-      textAlign: 'center',
-      border: 0,
-      '&::before, &::after': {
-        content: '""',
-        alignSelf: 'center',
       },
-    }),
-  }),
-  ({ theme, ownerState }) => ({
-    ...(ownerState.children &&
-      ownerState.orientation !== 'vertical' && {
+    },
+    {
+      props: {
+        orientation: 'vertical',
+      },
+      style: {
+        height: '100%',
+        borderBottomWidth: 0,
+        borderRightWidth: 'thin',
+      },
+    },
+    {
+      props: {
+        flexItem: true,
+      },
+      style: {
+        alignSelf: 'stretch',
+        height: 'auto',
+      },
+    },
+    {
+      props: ({ ownerState }) => !!ownerState.children,
+      style: {
+        display: 'flex',
+        whiteSpace: 'nowrap',
+        textAlign: 'center',
+        border: 0,
+        '&::before, &::after': {
+          content: '""',
+          alignSelf: 'center',
+        },
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.children && ownerState.orientation !== 'vertical',
+      style: {
         '&::before, &::after': {
           width: '100%',
           borderTop: `thin solid ${(theme.vars || theme).palette.divider}`,
         },
-      }),
-  }),
-  ({ theme, ownerState }) => ({
-    ...(ownerState.children &&
-      ownerState.orientation === 'vertical' && {
-        flexDirection: 'column',
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.orientation === 'vertical' && ownerState.children,
+      style: {
         '&::before, &::after': {
+          flexDirection: 'column',
```

Comment:
@sai6855 `flexDirection` was out of psuedo selector, could you recheck if this is the correct change?

Reply:
Opps sorry, fixed in this commit https://github.com/mui/material-ui/pull/41366/commits/0b39bba5d6ba2080aa85319e8f4b25802742aeca